src/icp_canister_backend/tests/rewards_tests.rs

```diff
@@ -90,73 +87,431 @@ fn test_reward_rate_increase_decrease_during_episodes() {
+    
+    let initial_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    let current_episode = initial_time / EPISODE_DURATION;
+    let last_reward_episode = (initial_time + EPISODE_DURATION * 12) / EPISODE_DURATION;
+    let exact_reward_duration = (last_reward_episode + 1) * EPISODE_DURATION - initial_time;
+    
+    let rewards_before = pic
+        .query_call(canister_id, user, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get rewards before transition");
+    
+    let next_episode_start = (current_episode + 1) * EPISODE_DURATION;
+    let time_to_next_episode = next_episode_start - initial_time;
+    advance_time(&pic, time_to_next_episode + 1);
+    
+
+    // Get rewards after episode transition
+    let rewards_after_transition = pic
+        .query_call(canister_id, user, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get rewards after transition");
+    
+    // Calculate expected reward increase during episode transition
+    let time_elapsed = time_to_next_episode + 1;
+    let reward_rate = (reward_amount.clone() * icp_canister_backend::PRECISION_SCALE.clone()) / Nat::from(exact_reward_duration);
+    let expected_increase = (reward_rate.clone() * Nat::from(time_elapsed)) / icp_canister_backend::PRECISION_SCALE.clone();
+    let actual_increase = rewards_after_transition.clone() - rewards_before.clone();
+    assert_with_error!(&actual_increase, &expected_increase, &ALLOWED_ERROR, "Reward increase during episode transition");
+    
+    // Advance through several more episodes
+    for _ in 0..3 {
+        advance_time(&pic, EPISODE_DURATION);
+    }
+    
+    // Get final rewards after multiple episode transitions
+    let final_rewards = pic
+        .query_call(canister_id, user, "get_deposits_rewards", encode_args((vec![0u64],)).unwrap())
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get final rewards");
+    
+    // Calculate expected additional rewards from 3 more episodes
+    let additional_time = EPISODE_DURATION * 3;
+    let expected_additional = (reward_rate.clone() * Nat::from(additional_time)) / icp_canister_backend::PRECISION_SCALE.clone();
+    let expected_final = rewards_after_transition.clone() + expected_additional;
+    assert_with_error!(&final_rewards, &expected_final, &ALLOWED_ERROR, "Final rewards after multiple episodes");
+    
+    // Calculate expected portion of total reward based on time elapsed
+    let total_time_elapsed = time_to_next_episode + 1 + (EPISODE_DURATION * 3);
+    let expected_portion = (reward_amount.clone() * Nat::from(total_time_elapsed)) / Nat::from(exact_reward_duration);
+    assert_with_error!(&final_rewards, &expected_portion, &ALLOWED_ERROR, "Reward portion calculation");
 }
+
+#[test]
+fn test_mathematical_precision_large_numbers() {
```

Comment:
Also, I don't understand the aim of the test.
Create another test to check if the rewards work with small amounts.
One deposit should be big, like 10 BTC, another small, like 100$ in BTC, and a reward of 1$ in BTC. Assume the Bitcoin price is 100k.